ambry-store/src/test/java/com.github.ambry.store/IndexValueTest.java

```diff
@@ -108,14 +170,124 @@ private void verifyIndexValue(IndexValue value, String logSegmentName, long size
-    assertEquals("ExpiresAtMs not as expected", expiresAtMs, value.getExpiresAtMs());
+    if (version == 0) {
+      assertEquals("ExpiresAtMs not as expected", expiresAtMs, value.getExpiresAtMs());
+      assertEquals("Operation time mismatch", Utils.Infinite_Time, value.getOperationTimeInMs());
+      assertEquals("ServiceId mismatch ", IndexValue.SERVICE_CONTAINER_ID_DEFAULT_VALUE, value.getServiceId());
+      assertEquals("ContainerId mismatch ", IndexValue.SERVICE_CONTAINER_ID_DEFAULT_VALUE, value.getContainerId());
+    } else {
+      assertEquals("ExpiresAtMs not as expected",
+          expiresAtMs != Utils.Infinite_Time ? (expiresAtMs / Time.MsPerSec) * Time.MsPerSec : Utils.Infinite_Time,
+          value.getExpiresAtMs());
+      assertEquals("Operation time mismatch",
+          operationTimeInMs != Utils.Infinite_Time ? (operationTimeInMs / Time.MsPerSec) * Time.MsPerSec
+              : Utils.Infinite_Time, value.getOperationTimeInMs());
+      assertEquals("ServiceId mismatch ", serviceId, value.getServiceId());
+      assertEquals("ContainerId mismatch ", containerId, value.getContainerId());
+    }
     assertEquals("Original message offset not as expected", originalMessageOffset, value.getOriginalMessageOffset());
   }
+
+  /**
+   * Verifies that construction of {@link IndexValue} fails with an invalid {@link ByteBuffer} value
+   * @param value the source {@link IndexValue} to contruct the bad one
+   * @param logSegmentName the log segment name to be used to construct the {@link IndexValue}
+   */
+  private void verifyInvalidValueSize(IndexValue value, String logSegmentName) {
+    try {
+      int capacity = TestUtils.RANDOM.nextInt(value.getBytes().capacity());
+      ByteBuffer invalidValue = ByteBuffer.allocate(capacity);
+      invalidValue.put(value.getBytes().array(), 0, capacity);
+      new IndexValue(logSegmentName, invalidValue, version);
+      fail(
+          "Contruction of IndexValue expected to fail with invalid byte buffer capacity of " + invalidValue.capacity());
+    } catch (IllegalArgumentException e) {
+    }
+  }
+
+  /**
+   * Constructs IndexValue based on the args passed and for the given version
+   * @param size the size of the blob that this index value refers to
+   * @param offset the {@link Offset} in the {@link Log} where the blob that this index value refers to resides
+   * @param expirationTimeInMs the expiration time in ms at which the blob expires
+   * @param operationTimeInSecs operation time of the entry
+   * @param serviceId the serviceId that this blob belongs to
+   * @param containerId the containerId that this blob belongs to
+   * @param version the version with which to construct the {@link IndexValue}
+   * @return the {@link IndexValue} thus constructed
+   */
+  static IndexValue getIndexValue(long size, Offset offset, long expirationTimeInMs, long operationTimeInSecs,
```

Comment:
3rd one's purpose is diff from 1st and 2nd. So ruling out the 3rd, we could merge 1st and 2nd. But then, from the caller's perspective, if interested in Version_0, it might be confusing if the caller is expected to pass in operation time, serviceId and containerId. Having this in place, makes the test more readable.

for eg check out the difference between these two calls: 
IndexValueTest.getIndexValue(CuratedLogIndexState.PUT_RECORD_SIZE, currentEndOffset, state.time.seconds(),  PersistentIndex.VERSION_0));

If we merge 1st and 2nd method to one:
IndexValueTest.getIndexValue(CuratedLogIndexState.PUT_RECORD_SIZE, currentEndOffset, state.time.seconds(), Utils.InfiniteTime, IndexValue.SERVICE_CONTAINER_ID_DEFAULT_VALUE, IndexValue.SERVICE_CONTAINER_ID_DEFAULT_VALUE, PersistentIndex.VERSION_0));